### ISSUE TITLE
Guild (X) DT tooltip doesn't display

### ISSUE THREAD

**Issue by Fyurie** [Chalsean/XFaction#310]
I'm not LUA-savvy enough to tell if this error is due to a bug in XFaction or ElvUI (or both/neither!), so I figured I'd post this trace here for investigation.

As for the issue itself: I have a Guild (X) DT enabled, and hovering over it displays nothing, instead generating an instance of this error. The Link (X) DT seems to function fine, however.

```
Interface/AddOns/XFaction/DataText/DTGuild.lua:219: attempt to index field '?' (a nil value)
[string "@Interface/AddOns/XFaction/DataText/DTGuild.lua"]:219: in function `OnEnter'
[string "@Interface/AddOns/XFaction/DataText/DTGuild.lua"]:29: in function `OnEnter'
[string "@Interface/AddOns/ElvUI/Core/Modules/DataTexts/DataTexts.lua"]:237: in function `func'
[string "@Interface/AddOns/ElvUI/Core/Modules/DataTexts/DataTexts.lua"]:127: in function <...ce/AddOns/ElvUI/Core/Modules/DataTexts/DataTexts.lua:117>

Locals:
self = <table> {
 __name = "DTGuild"
 regularFont = regularFont {
 }
 key = "/o]Ic&A7a4w"
 initialized = true
 headerFont = headerFont {
 }
 tooltip = Frame {
 }
 isReverseSort = false
 ldbObject = <table> {
 }
}
this = MinimapPanel_DataText1 {
 MouseEnters = <table> {
 }
 PixelSnapDisabled = true
 text = FontString {
 }
 name = "Guild (X)"
 db = <table> {
 }
 pointIndex = 1
 parentName = "MinimapPanel"
 0 = <userdata>
 MouseLeaves = <table> {
 }
 parent = MinimapPanel {
 }
 battleStats = false
 overlay = Texture {
 }
}
orderEnabled = <table> {
 1 = <table> {
 }
 3 = <table> {
 }
 2 = <table> {
 }
 5 = <table> {
 }
 4 = <table> {
 }
 6 = <table> {
 }
 9 = <table> {
 }
 8 = <table> {
 }
 10 = <table> {
 }
}
(for index) = 7
(for limit) = 9
(for step) = 1
i = 7
(*temporary) = <function> defined @Interface/AddOns/ElvUI_SLE/libs/LibQTip-1.0/LibQTip-1.0.lua:634
(*temporary) = Frame {
 0 = <userdata>
 OnBackdropLoaded = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:12
 GetBackdropColor = <function> defined @Interface/SharedXML/SharedTooltipTemplates.lua:191
 key = "DTGuild"
 SetBackdropBorderColor = <function> defined @Interface/SharedXML/SharedTooltipTemplates.lua:195
 anchorframe = LibDBIcon10_SavedInstances {
 }
 scrollChild = Frame {
 }
 ApplyBackdrop = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:22
 SetBackdrop = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:27
 ClearBackdrop = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:34
 cell_margin_h = 6
 GetBackdropBorderColor = <function> defined @Interface/SharedXML/SharedTooltipTemplates.lua:199
 GetBackdrop = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:19
 height = 2
 SetBorderBlendMode = <function> defined @Interface/SharedXML/SharedTooltipTemplates.lua:203
 regularFont = GameTooltipText {
 }
 layoutType = "TooltipDefaultLayout"
 cell_margin_v = 3
 headerFont = GameTooltipHeaderText {
 }
 lines = <table> {
 }
 SetupTextureCoordinates = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:16
 GetBackdropCoordValue = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:15
 HasBackdropInfo = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:18
 columns = <table> {
 }
 scrollFrame = ScrollFrame {
 }
 GetEdgeSize = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:14
 NineSlice = Frame {
 }
 isSkinned = true
 colspans = <table> {
 }
 TooltipBackdropOnLoad = <function> defined @Interface/SharedXML/SharedTooltipTemplates.lua:172
 width = 30
 backdrop = Frame {
 }
 labelProvider = <table> {
 }
 SetupPieceVisuals = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:17
 SetBackdropColor = <function> defined @Interface/SharedXML/SharedTooltipTemplates.lua:187
 OnBackdropSizeChanged = <function> defined @Interface/AddOns/Blizzard_Deprecated/Deprecated_9_1_5.lua:13
}
(*temporary) = nil
(*temporary) = "7"
(*temporary) = "7"
(*temporary) = <table> {
 6 = Frame {
 }
 2 = Frame {
 }
 3 = Frame {
 }
 1 = Frame {
 }
 4 = Frame {
 }
 5 = Frame {
 }
}
(*temporary) = Frame {
 0 = <userdata>
 OnBackdropLoaded = <function> defined @Interface/SharedXML/Backdrop.lua:152
 GetBackdropColor = <function> defined @Interface/SharedXML/Backdrop.lua:390
 SetupTextureCoordinates = <function> defined @Interface/SharedXML/Backdrop.lua:214
 OnBackdropSizeChanged = <function> defined @Interface/SharedXML/Backdrop.lua:182
 HasBackdropInfo = <function> defined @Interface/SharedXML/Backdrop.lua:278
 SetBackdropBorderColor = <function> defined @Interface/SharedXML/Backdrop.lua:422
 GetEdgeSize = <function> defined @Interface/SharedXML/Backdrop.lua:188
 ApplyBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:294
 SetBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:329
 ClearBackdrop = <function> defined @Interface/SharedXML/Backdrop.lua:282
 GetBackdropBorderColor = <function> defined @Interface/Shar```

**Comment by waggz81** [comment]
I had this issue too, deleting my XFaction.lua files fixed it, but Chalsean mentioned you should be able to edit the columns in XFaction settings to reset the configuration and fix it as well.

**Comment by Fyurie** [comment]
This worked for me. Thanks.

Should this be closed, or would Chalsean prefer to look into the root issue before this vanishes?

**Comment by Chalsean** [comment]
ill look into.  worst case probably just bundle a release that resets Guild (X) settings

**Comment by Chalsean** [comment]
its from the covenant stuff getting removed i think, it left settings in place

**Comment by Chalsean** [comment]
Fixed in 4.1.14
